modules/nf-core/decoupler/progeny/main.nf

```diff
@@ -0,0 +1,119 @@
+process DECOUPLER_PROGENY {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/7d/7d304d1b25aa80ce8c44c8d34ea45d6b4f6f50697a4effcf1b9be4a54db19928/data' :
+        'community.wave.seqera.io/library/decoupler-py_matplotlib_pandas_pip_pruned:0d4681dad9987ec5' }"
+
+    input:
+    tuple val(meta), path(mat)
+
+    output:
+    tuple val(meta), path("progeny_mlm_estimate_decoupler.tsv") , emit: dc_estimate
+    tuple val(meta), path("progeny_mlm_pvals_decoupler.tsv")    , emit: dc_pvals
+    tuple val(meta), path("acts_decoupler.tsv")                 , emit: dc_acts
+    tuple val(meta), path("figures/*umap.png")                  , emit: dc_umap
+    tuple val(meta), path("figures/*violin.png")                , emit: dc_violin
+    tuple val(meta), path("matrixplot_decoupler.png")           , emit: dc_matrixplot
+    path("versions.yml"), emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ? new groovy.json.JsonSlurper().parseText(task.ext.args) : [:]
+    def organism = args.organism ?: 'human'
+    def top_genes = args.top_genes ?: 1000
```

Comment:
There are no guidelines on this. Does this come under the same realm as custom meta fields. i.e. should these be `input:` ?

Reply:
I did those as input in my first design :) but then I remembered this one: https://nf-co.re/docs/guidelines/components/modules#non-file-mandatory-command-arguments
and I changed it to `ext.args`.

Reply:
(I prefer them as `input:`, though. It's a lot cleaner)